django/forms/fields.py

```diff
@@ -762,14 +763,19 @@ class URLField(CharField):
 
     def __init__(self, *, assume_scheme=None, **kwargs):
         if assume_scheme is None:
-            warnings.warn(
-                "The default scheme will be changed from 'http' to 'https' in Django "
-                "6.0. Pass the forms.URLField.assume_scheme argument to silence this "
-                "warning.",
-                RemovedInDjango60Warning,
-                stacklevel=2,
-            )
-            assume_scheme = "http"
+            if settings.FORMS_URLFIELD_ASSUME_HTTPS:
+                assume_scheme = "https"
+            else:
+                warnings.warn(
+                    "The default scheme will be changed from 'http' to 'https' in "
+                    "Django 6.0. Pass the forms.URLField.assume_scheme argument to "
+                    "silence this warning, or set the FORMS_URLFIELD_ASSUME_HTTPS "
+                    "transitional setting to True to opt into using 'https' as the new "
+                    "default scheme.",
+                    RemovedInDjango60Warning,
+                    stacklevel=2,
+                )
+                assume_scheme = "http"
```

Comment:
If we do this the other way round, so go with `https` **unless** (it would have to be) `settings.FORMS_URLFIELD_ASSUME_HTTP` were set to `True` then we could have only the single deprecation warning, when checking overridden settings, and not have the second warning here. 

This would mean folks opting in to the future wouldn’t have any warnings, so no need to silence them.

Reply:
I don’t understand what you’re proposing. An HTTP setting that defaults to True, but if set to False opts in to the future? Why would that mean no warning when opting in?

Reply:
I mean:

1. Assume `https` (No warning)
2. Then a setting, which defaults `False`, but if `True` falls back to `http`. 

So _Opted-in to the future by default_, with a fallback. 

3. If the fallback is used, you get a warning that it's going away. 

Make sense?

Reply:
> Make sense? (It might not: I've been at the computer all day, so may well be hallucinating by this point 😜)

@carltongibson I'm against your idea, as it would mean that we're making a backward incompatible change that user will probably not notice.

Reply:
Yeah... it's only looking at it that it becomes 100% clear, but I'd lean to the smoother path **for updating** where's there's some choice. 

As I said, I won't object if we don't go that way, but I wanted to make sure we'd considered it. 

🎁

Reply:
Posts crossed. 

@felixxm OK, you're the boss 😉